lib/compiler/src/compiler.rs

```diff
@@ -36,6 +36,12 @@ pub trait CompilerConfig {
         // in case they create an IR that they can verify.
     }
 
+    /// Enable NaN canonicalization.
+    fn enable_nan_canonicalization(&mut self) {
+        // By default we do nothing, each backend will need to customize this
+        // in case they create an IR that they can verify.
+    }
```

Comment:
Receive another variable, and make the function not have a default implementation.

```suggestion
    fn enable_nan_canonicalization(&mut self, enable: bool);
```

Reply:
If it takes a bool to enable or disable, maybe `set_nan_canonicalization` would be preferred? Or `set_canonicalize_all_nans`?

Reply:
Yeah, perhaps naming can be better in that case!

Reply:
For the Wasm features we simply put the name directly (without set/enable).
https://github.com/wasmerio/wasmer/blob/master/lib/wasmer-types/src/features.rs#L107-L110

Perhaps
`canonicalize_nans(&mut self, enable: bool);` ?